### PR TITLE
Retry on network errors in Firefox and Safari too

### LINKED ISSUE #41: Cross browser handling of retrying on network connection issues
Based on [this issue](https://github.com/sindresorhus/p-retry/issues/37), p-retry now can be used to retry based on network errors on Chrome. Nice!

However, as mentioned in the 'slightly relevant ' [fetch issue](https://github.com/whatwg/fetch/issues/526), different browsers have different messages based on the network errors:
Chrome: TypeError: Failed to fetch
Firefox: TypeError: NetworkError when attempting to fetch resource.
Safari: TypeError: The Internet connection appears to be offline.
([source](https://gist.github.com/keeth/d5d94dab40b4d697618c075e155197da))

Would it be possible to support network errors in Firefox and Safari, and also retry on them -- also happy to put in a PR by referencing https://github.com/sindresorhus/p-retry/pull/38/files.

### PR DESCRIPTION
**Pull request by peterj35**
Fixes #41

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -39,7 +39,8 @@ const run = async () => {
...
 
 
-It doesn't retry on `TypeError` as that's a user error. The only exclusion to this logic is when `TypeError` is thrown by `fetch`'s API with the message 'Failed to fetch', [which indicates that a request was not successful due to a network error](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful). However, beware that `fetch` may throw `TypeError` with different error messages on different platforms for similar situations. See [whatwg/fetch#526 (comment)](https://github.com/whatwg/fetch/issues/526#issuecomment-554604080)."  
+Does not retry on most `TypeErrors`, with the exception of network errors. This is done on a best case basis as different browsers have different [messages](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful) to indicate this. See [whatwg/fetch#526 (comment)](https://github.com/whatwg/fetch/issues/526#issuecomment-554604080)
```
Comment (sindresorhus) [inline readme.md:42]: You need to update index.d.ts too.

File: index.js [hunk 2]
```diff
@@ -1,6 +1,12 @@
 'use strict';
 const retry = require('retry');
 
+const networkErrorMsgs = [
+	'Failed to fetch', // Chrome
+	'NetworkError when attempting to fetch resource', // FF
```
Comment (sindresorhus) [inline index.js:6]:
```suggestion
	'NetworkError when attempting to fetch resource', // Firefox
```

File: readme.md (hunk 1 above)
Comment (peterj35) [reply readme.md:42]: Thanks, done

File: index.js (hunk 2 above)
Comment (peterj35) [reply index.js]: Implemented

**Comment by sindresorhus** [comment]
CI is failing

**Pull request merged**